app/src/main/java/com/futurice/freesound/feature/search/DefaultSearchDataModel.java

```diff
@@ -37,33 +38,33 @@
     private final FreeSoundSearchService freeSoundSearchService;
 
     @NonNull
-    private final BehaviorRelay<Option<List<Sound>>> lastResults = BehaviorRelay.create();
+    private final BehaviorSubject<Option<List<Sound>>> lastResults = BehaviorSubject.create();
 
     DefaultSearchDataModel(@NonNull final FreeSoundSearchService freeSoundSearchService) {
         this.freeSoundSearchService = get(freeSoundSearchService);
     }
 
     @Override
     @NonNull
-    public Observable<Unit> querySearch(@NonNull final String query) {
+    public Single<Unit> querySearch(@NonNull final String query) {
         return freeSoundSearchService.search(get(query))
                                      .map(SoundSearchResult::results)
                                      .map(Option::ofObj)
-                                     .doOnNext(lastResults)
-                                     .map(Unit::asUnit);
+                                     .doOnSuccess(lastResults::onNext)
+                                     .map(it -> Unit.DEFAULT);
     }
 
     @Override
     @NonNull
     public Observable<Option<List<Sound>>> getSearchResults() {
-        return lastResults.asObservable();
+        return lastResults;
     }
 
     @Override
     @NonNull
-    public Observable<Unit> clear() {
-        return Observable.just(Option.<List<Sound>>none())
-                         .doOnNext(lastResults)
-                         .map(__ -> Unit.DEFAULT);
+    public Single<Unit> clear() {
```

Comment:
This is probably out of scope for this change, but can you think of a reason why `clear` needs to return an `Observable` or a `Single`? Could we just call `lastResults.onNext(none())` directly and return `void`?